Robot2018/src/org/usfirst/frc/team199/Robot2018/RobotMap.java

```diff
@@ -178,4 +178,15 @@ public int getPort(String key, int def) {
 		return (int) SmartDashboard.getNumber("Port/" + key, def);
 	}
 
+	/**
+	 * Uses SmartDashboard and math to calculate a *great* default kD
+	 */
+	public double calcDefkD() {
+		double timeConstant = Robot.getConst("Omega Max", 5330) * Robot.getConst("Mass of Robot", 54.4311)
+				* Robot.getConst("Radius of Drivetrain Wheel", 0.0635)
+				* Robot.getConst("Radius of Drivetrain Wheel", 0.0635) / Robot.getConst("Stall Torque", 2.41);
```

Comment:
The calculations we did were based on the Omega Max and Stall Torque at the *wheel*, not the motor. So they depend on the gear ratio in addition to the value for the motors. As a result, you need to factor the current gear ratio into your calcs.

Reply:
Done in latest commit in latest PR.